File: 2ºCARRERA/tprog/practica2_872815_873338/java/Add.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Add.java
 */

import java.util.Stack;

class Add implements Instruccion
{
    public int ejecuta(Stack<Integer> pila, int pc) {
        int num1, suma;
        suma = pila.pop();
        suma = suma + pila.pop();
        pila.push(suma);
        pc++;
        return pc;
    }
    public void muestra() {
        System.out.print("add");
    }
}

File: 2ºCARRERA/tprog/practica2_872815_873338/c++/add.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : add.cc
 */

#include "add.h"

void Add::ejecuta(stack<int32_t> &pila, int& pc) {
    int32_t num1,suma;
    suma = pila.top();
    pila.pop();
    suma += pila.top();
    pila.pop();
    pila.push(suma);
    pc++;
}

void Add::muestra(){
    cout << "add";
}